pkg/kube/translation/apisix_route.go

```diff
@@ -304,6 +346,24 @@ func (t *translator) translateRouteMatchExprs(nginxVars []configv2alpha1.ApisixR
 	return vars, nil
 }
 
+// translateTCPRouteNotStrictly is not strictly translation, only generate ID and Name for delete Event.
+func (t *translator) translateTCPRouteNotStrictly(ctx *TranslateContext, ar *configv2alpha1.ApisixRoute) error {
+	for _, part := range ar.Spec.TCP {
+		backend := &part.Backend
+		sr := apisixv1.NewDefaultStreamRoute()
+		name := apisixv1.ComposeStreamRouteName(ar.Namespace, ar.Name, part.Name)
+		sr.ID = id.GenID(name)
+		sr.ServerPort = part.Match.IngressPort
+		ups, err := t.translateUpstreamNotStrictly(ar.Namespace, backend.ServiceName, backend.Subset, backend.ServicePort.IntVal)
+		if err != nil {
+			return err
+		}
+		sr.Upstream = ups
+		ctx.addStreamRoute(sr)
```

Comment:
Should also add the `ups` to `ctx`.

Reply:
ref to：https://github.com/apache/apisix-ingress-controller/blob/master/pkg/kube/translation/apisix_route.go#L333
Are you sure need to add `ups` to `ctx`?

Reply:
We should.

By the way, the code in your link should also be changed as now stream_route can use upstream_id.

Reply:
Fixed.